core/src/com/fourinachamber/fortyfive/map/shop/ShopScreenController.kt

```diff
@@ -68,10 +68,9 @@ class ShopScreenController(onj: OnjObject) : ScreenController() {
 
 
         val backButton = onjScreen.namedActorOrError(backButtonName)
-        backButton.onClick { shopWidgetNames.forEach{
-            (screen.namedActorOrError(it) as ShopWidget).giveResourcesBack()
+        backButton.onClick {
             personWidget.giveResourcesBack()
```

Comment:
Not really sure but I think the better solution here would be for personWidget to extend Disposable and to use the `.dispose` function instead